nginx_stage/lib/nginx_stage.rb

```diff
@@ -145,6 +145,19 @@ def self.nginx_env(user:)
     }.merge(pun_custom_env)
   end
 
+  # Replace environment with specified environment
+  # @param env [Hash] the new environment
+  # @return [ENV] the reset environment
+  def self.nginx_env_reset(env:)
```

Comment:
Can we actually just replace `nginx_env` with this implementation and rename it to `clean_nginx_env` (along with comments about what it does)?

I feel like this was the expectation all along - that the PUN boot with a very specific ENV, no more no less, but we never looked closely at open3's APIs. 

Note that several places call `nginx_env` so renaming it's going to create a couple other changes.

Reply:
Calling `clean_nginx_env` is would then be mutative, but at least it'd be idempotent.

Reply:
I thought that might be an issue due to this that is used in template:
https://github.com/OSC/ondemand/blob/88e868a8947f6397b0542d96151576c69dd6d2bf/nginx_stage/lib/nginx_stage/views/pun_config_view.rb#L128-L130

But ENV.replace returns the new ENV so seems like your idea should be fine. I went ahead and made the necessary adjustments.